msm/skill_repo.py

```diff
@@ -45,24 +46,40 @@ def read_file(self, filename):
         with open(join(self.path, filename)) as f:
             return f.read()
 
-    def update(self):
+    def __prepare_repo(self):
         if not exists(dirname(self.path)):
             makedirs(dirname(self.path))
 
-        with git_to_msm_exceptions():
-            if not isdir(self.path):
-                Repo.clone_from(self.url, self.path)
+        if not isdir(self.path):
+            Repo.clone_from(self.url, self.path)
 
-            git = Git(self.path)
-            git.config('remote.origin.url', self.url)
-            git.fetch()
+        git = Git(self.path)
+        git.config('remote.origin.url', self.url)
+        git.fetch()
 
         try:
             git.checkout(self.branch)
             git.reset('origin/' + self.branch, hard=True)
         except GitCommandError:
             raise MsmException('Invalid branch: ' + self.branch)
 
+    def update(self):
+        with git_to_msm_exceptions():
```

Comment:
`git_to_msm_exception` is redundant since `GitError` is already caught

Reply:
If the final attempt fails it would raise a GitError that wouldn't be caught, shouldn't this be left for that case?

Reply:
Oh yeah, didn't realize that. Would it make sense to place it only around the final `self.__prepare_repo` call since that's the only thing it applies to?

Reply:
That makes sense. I'll change that.